code/go/0chain.net/miner/chain_test.go

```diff
@@ -0,0 +1,390 @@
+package miner
```

Comment:
chain.go looks like a difficult to write a unit test for. 

Pick a method and focus on testing only the input and outputs of the method. Remeber that chain satisfies the [Entity](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/core/datastore/entity.go#L19), [Stoppable](https://pkg.go.dev/github.com/rcrowley/go-metrics#Stoppable), [CreateTrackable](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/core/datastore/creation_date.go#L9) and [Chainer](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/chaincore/chain/block_fetcher.go#L316) interfaces, so if a test fucntion calls any of these interface methods you can mock the response.